lib/chef/resource/launchd.rb

```diff
@@ -0,0 +1,285 @@
+#
+# Author:: Mike Dodge (<mikedodge04@gmail.com>)
+# Copyright:: Copyright (c) 2015 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/resource/file'
+require 'chef/provider/launchd'
+require 'chef/mixin/securable'
+
+class Chef
+  class Resource
+    class Launchd < Chef::Resource::File
+      include Chef::Mixin::Securable
+
+      provides :launchd
+
+      def initialize(name, run_context = nil)
+        super
+        @provider = Chef::Provider::Launchd
+        @resource_name = :launchd
+        @action = 'create'
+        @cookbook = nil
+        @source = nil
+        @label = name
+        @disabled = false
+        @service_type = 'system'
+        @path = "/Library/LaunchDaemons/#{name}.plist"
+      end
+
+      def label(label = @label)
```

Comment:
I'm pretty sure you want `def label(label=nil)` here. set_or_return will return the value of the instance variable if you pass nil, or set it if you pass non-nil. In this case, any time you call `label` without arguments you're going to be resetting it to itself (and returning that value), which is unnecessary.

This example might help:

```
require 'chef/mixin/params_validate'

include Chef::Mixin::ParamsValidate

@val_nil = "initialized value"
@val_passed = "initialized value"

def val_nil(arg=nil)
  set_or_return(:val_nil, arg, :kind_of => String)
end

def val_passed(arg = "method argument")
  set_or_return(:val_passed, arg, :kind_of => String)
end

puts val_nil
puts val_passed

puts val_nil("set")
puts val_passed("set")

puts val_nil
puts val_passed
```

```
$ ruby set_or_return.rb
initialized value
method argument
set
set
set
method argument
```